internal/app/1Click/on-premise/check_depedencies.go

```diff
@@ -0,0 +1,31 @@
+package onpremise
+
+import (
+	"os/exec"
+
+	"github.com/NethermindEth/1Click/configs"
+	log "github.com/sirupsen/logrus"
+)
+
+/*
+CheckDependencies :
+This function is responsible for checking if on-premise setup dependencies are installed on host machine
+
+params :-
+a. dependencies []string
+List of dependencies to be checked
+
+returns :-
+a. error
+Error if any
+*/
+func CheckDependencies(dependencies []string) (pending []string) {
```

Comment:
Shouldn't it be bound to a struct ?

Reply:
If we get dependencies from a config file then we need a struct for them. Will refactor this. It will be good if we discuss first what is the best way to handle our configuration values to avoid hardcoding them.

Reply:
Refactoring would be cool and we can have a discussion for sure